test/benchmark_http_client_test.cc

```diff
@@ -310,8 +343,46 @@ TEST_F(BenchmarkClientHttpTest, BadContentLength) {
   };
 
   EXPECT_CALL(stream_encoder_, encodeData(_, _)).Times(0);
-  testBasicFunctionality(1, 1, 1);
+
+  TestBenchmarkClientProcessesExpectedInflightRequests(1, 1, 1, request_generator);
   EXPECT_EQ(1, getCounter("http_2xx"));
 }
 
+TEST_F(BenchmarkClientHttpTest, RequestGeneratorProvidingDifferentPathsSendsRequestsOnThosePaths) {
+  std::vector<HeaderMapPtr> requests_for_generator_to_send;
+  const std::initializer_list<std::pair<std::string, std::string>> header_map_for_first_request{
+      {":scheme", "http"},
+      {":method", "GET"},
+      {":path", "/a"},
+      {":host", "localhost"},
+      {"Content-Length", "1313"}};
+  const std::initializer_list<std::pair<std::string, std::string>> header_map_for_second_request{
+      {":scheme", "http"},
+      {":method", "GET"},
+      {":path", "/b"},
+      {":host", "localhost"},
+      {"Content-Length", "1313"}};
+  requests_for_generator_to_send.push_back(
+      std::make_shared<Envoy::Http::TestRequestHeaderMapImpl>(header_map_for_first_request));
+  requests_for_generator_to_send.push_back(
+      std::make_shared<Envoy::Http::TestRequestHeaderMapImpl>(header_map_for_second_request));
+  std::vector<HeaderMapPtr>::iterator request_iterator;
+  request_iterator = requests_for_generator_to_send.begin();
+  RequestGenerator request_generator = [&request_iterator]() {
+    return std::make_unique<RequestImpl>(*request_iterator++);
+  };
+  absl::flat_hash_set<std::string> expected_requests;
+  expected_requests.insert(
+      getTestRecordedProperty(Envoy::Http::TestRequestHeaderMapImpl(header_map_for_first_request)));
+  expected_requests.insert(getTestRecordedProperty(
+      Envoy::Http::TestRequestHeaderMapImpl(header_map_for_second_request)));
+
+  EXPECT_CALL(stream_encoder_, encodeData(_, _)).Times(2);
+
+  // Most of the testing happens inside of this call. Will confirm that the requests received match
+  // the expected requests vector.
+  TestBenchmarkClientProcessesExpectedInflightRequests(1, 1, 2, request_generator,
```

Comment:
probably out of scope, but now that we have c++17 it looks like we could potentially
have named arguments [1], which could enhance readability of the numbers we pass to these calls. 

[1] https://www.fluentcpp.com/2018/12/14/named-arguments-cpp/

Reply:
I refactored this to reduce the number of arguments instead. Figured that this would be cleaner.